osf_tests/factories.py

```diff
@@ -505,43 +507,68 @@ class Meta:
         model = models.PreprintProvider
 
 
+def sync_set_identifiers(preprint):
+    ezid_return_value ={
+        'response': {
+            'success': '{doi}osf.io/{guid} | {ark}osf.io/{guid}'.format(
+                doi=settings.DOI_NAMESPACE, ark=settings.ARK_NAMESPACE, guid=preprint._id
+            )
+        },
+        'already_exists': False
+    }
+    preprint.set_preprint_identifiers(ezid_return_value)
```

Comment:
It's not intuitive for `set_preprint_identifiers` to receive the EZID response data. The model shouldn't really need to know anything about EZID at all. I'd suggest either getting rid of or modifying set_preprint_identifiers.

```python
identifiers = parse_identifiers(ezid_response)
preprint.set_identifier_value('doi', identifiers['doi'])
preprint.set_identifier_value('ark', identifiers['ark'])
```

Then you could avoid having to hand-write the response data in this test, and you get rid of EZID-parsing code in the model.